modules/apps/portal-search/portal-search-admin-web/src/main/resources/content/Language.properties

```diff
@@ -14,8 +14,10 @@ javax.portlet.title.com_liferay_portal_search_admin_web_portlet_SearchAdminPortl
 no-active-connections=No Active Connections
 no-search-engine-detected-help=No search engine detected. To determine why, please check the application logs. If necessary, reactivate the default search engine (if manually disabled) or install a search engine.
 nodes-x=Nodes ({0})
-reindex-all-search-indexes=Reindex all search indexes.
-reindex-all-spell-check-indexes=Reindex all spell check indexes.
+reindex-additional-indexes=Reindex application and developer indexes.
```

Comment:
@lipusz @rbohl you guys have any input on these keys? kind of did it quickly

Reply:
Thinking about this one:

> reindex-additional-indexes-help=Reindex components implementing the IndexReindexer service.

True, but might not get through to an admin. I'll try to come up with admin-frinedly text.

Reply:
First try:

> reindex-additional-indexes-help=Reindex data for applications and services that manage independent indexes. This action drops and recreates indexes managed by the IndexReindexer service.

Reply:
"This action drops and recreates indexes". Thats not necessarily true... IndexReindexer is an SPI interface, so its an extension point and will do whatever a developer implements... the RankingIndexReindexer does drop/recreate the rankings indexes, but thats just how we handle it for rankings

Reply:
reindex-additional-indexes-help=Reindex data for applications and services that manage independent indexes. Execution details for each index are determined by the IndexReindexer service implementation.

Reply:
These are still confusing rather than helping:
* admins are not supposed to know what is "application-driven" and what is not
* admins are not supposed to know which applications implement the SPI
* we provide no connection to end-user features that admins are most probably familiar with.

So I think, the UI should list each of the application names (like Result Rankings) that this button will reindex.

Would it be possible to register the app name or something that refer to the component through a property for example added to the component that we could track and associate with the reindexer to display on the UI?

Reply:
One caveat: in the search-admin-web module we would not be able to translate the names of the components into their respective UI translation as those lang keys are provided by different modules. :thinking:

Reply:
I'm still thinking about it, just jotting here my ideas (without an order for now):
* Redesign the "Index Actions" UI: split it into 3 sections (System & Virtual Instances Indexes, Application-driven Indexes, Entities) and RR and SS would appear under the "App-driven" category where the category itself would also have a "Reindex All" button. This would be yet another button for admins to push, though there would be a way to trigger reindexing a particular app index only (not all). Plus, with the changes from https://issues.liferay.com/browse/LPS-130753 there would be a complete control over the "scope".
* Do not add a new button: instead, when a "Reindex all search indexes" is invoked, we'd also reindex the app-driven indexes as well automatically. The downsize is that there would be no way for admins to reindex only RR or SS.
* Instead of changing the Search Admin, add a new Settings view to the RR and SS admin views, similar to how th Workflow Metrics does and add the reindex button there. Caveat: just like with WM, this UI is limited to a single Virtual Instance (not system-wise).

Thoughts?

@BryanEngler @arboliveira @rbohl

Reply:
This effort isn't supposed to get us to a LOL/saas ready UI, is it?

High level response to your ideas:

- I don't like the idea of doing search tuning indexing like workflow metrics. Seems like a real pain, as you pointed out, Tibor.

- I do think admins should be able to reindex everything that uses the DB as primary storage in one click.

- I also think admins should be able to index their search tuning indexes more granularly.

____
I don't know what's been discussed regarding how to skin the Virtual Instance scope cat for search in LOL so I might be lost, but Tibor made the mistake of at-mentioning me, so here goes. Everything in quotes is a button:

**[System Admin Screen]**
"Reindex All"
Re-indexes everything with primary storage in the DB.

"Reindex System Index"
Re-indexes `liferay-0`

[_If we might need to reindex just one, or a subset, of VIs:_ Add a field to enter the comma-separated companyIds, or, better, a Select from List with VI Names]
If blank reindex the below for all companyIds]

"Reindex Company Indexes"
"Reindex Application-Driven Indexes"
Reindex Individual Entities:
    "Reindex Journal Articles"
    "Reindex Blogs Entries"
    ...

Reply:
another thing to consider, the current reindex options (all or single entity) create background tasks. if we integrate the IndexReindexers into this functionality, should they also be put in background tasks?